### ISSUE TITLE
Cannot put Box inside of Text

### ISSUE THREAD

**Issue by goliney** [vadimdemedes/ink#344]
Hello, there. I'm in a process of migration to Ink v.3 (great job btw). 

So I had a component that had root `<Color>` element and a layout in it. Now when `Color` component is deprecated I replaced it with `Text`:

```js
export function Question({ message }) {
  return (
    <Text color="green">
      <Box marginRight={1}>
        <Text>?</Text>
      </Box>
      {message && (
        <Box marginRight={1}>
          <Text>{message}</Text>
        </Box>
      )}
    </Text>
  );
}
```

However, it throws an error:
![image](https://user-images.githubusercontent.com/2017344/88976182-4c13a400-d2c4-11ea-9bdf-3158e6afd559.png)

Apparently, you cannot put Box inside of the Text. 

As far as I can find a workaround for it, I was wondering if it something that will be changed.

**Comment by vadimdemedes** [comment]
Hey @goliney, my bad, I forgot to document that `<Box>` can't be nested inside `<Text>`, see https://github.com/vadimdemedes/ink/issues/337#issuecomment-667309544. Going to close this issue as a duplicate of https://github.com/vadimdemedes/ink/issues/337. Thanks for reporting!

**Comment by vadimdemedes** [comment]
To work around this, you'd need to assign `color="green"` to each `<Text>` component. I know it's less convenient, but because of this limitation, Ink's layout rendering is much more stable :)

**Comment by vadimdemedes** [comment]
Released a human readable error in [3.0.1](https://github.com/vadimdemedes/ink/releases/tag/v3.0.1).
